progress/progress.go

```diff
@@ -179,8 +188,15 @@ func (b *Bar) overwrite(prefix string) {
 	//  2 Move up one line
 	//  3 Clear to the end of the line
 	//  4 Print status text with optional prefix (spinner)
-	//  5 Print linebreak such that subsequent messaes print correctly.
-	fmt.Fprintf(b.out, "\r%v%v%v%v/%v %v\n", up, clear, prefix, b.index, b.total, b.text)
+	//  5 Print linebreak such that subsequent messages print correctly.
+	fmt.Fprintf(b.out, "\r%v%v%v%v", up, clear, prefix, b.format())
+}
+
+func (b *Bar) format() string {
+	if b.printWithStepCounter {
+		return fmt.Sprintf("%v/%v %v\n", b.index, b.total, b.text)
+	}
+	return fmt.Sprintf("%v\n", b.text)
 }
```

Comment:
If taking the suggestion above to give Bar a standard stringifier public method String, that would necessitate the minor tweak here of removing the trailing linebreaks:
```suggestion
func (b *Bar) String() string {
	if b.printWithStepCounter {
		return fmt.Sprintf("%v/%v %v", b.index, b.total, b.text)
	}
	return b.text
}
```